taichi/ir/type.h

```diff
@@ -193,6 +201,42 @@ class TensorType : public Type {
   Type *element_{nullptr};
 };
 
+class StructType : public Type {
+ public:
+  explicit StructType(std::vector<const Type *> elements)
+      : elements_(std::move(elements)) {
```

Comment:
Do we allow PointerType and QuantType in a StructType? Otherwise let's add a checker here?

Reply:
`PointerType` is allowed in a `StructType` for passing references. Though it is not yet tested, I think `QuantType` can be put into a struct naturally as well.

Reply:
Emmm it seems that we don't support `QuantType` yet... I'll add a checker then...